Sources/Lexer.swift

```diff
@@ -20,9 +27,7 @@ struct Lexer {
   func createToken(string: String, at range: Range<String.Index>) -> Token {
```

Comment:
I admit it's quite hard to review such code (on which I haven't been really working myself, so this review is quite out of context too) without any doc-comments on what the function is supposed to do.

At least a small doc-comment telling the preconditions (= it's expecting a `string` which already starts with `{` + one of the `tokenChars` and ends with the corresponding `tokenCharMap` + `}`) and what the `range` is supposed to mean here, would help.

_I mean… I'm assuming (from reading the implementation) that the check of the `string` starting with `{{/{%/{#`and ending with the corresponding closing sequence… is done in a separate method upstream, right? But I'm just guessing at that point_

Reply:
Yeah the checks happen in `Scanner.scanForTokenStart` and `Scanner.scanForTokenEnd`

This method receives a token's content, and creates a corresponding `.variable`, `.block`, `comment` or `.text` token. The range is used for the improved error functionality, so that the parser can point to where in the file the issue is.